cfy_manager/main.py

```diff
@@ -859,7 +880,50 @@ def configure(verbose=False,
 
     config[UNCONFIGURED_INSTALL] = False
     logger.notice('Configuration finished successfully!')
-    _finish_configuration()
+    _finish_configuration(components)
+
+
+def _all_main_services_removed(dir_path):
+    return all(not os.path.exists(os.path.join(dir_path, service_name))
+               for service_name in MAIN_SERVICES_NAMES)
+
+
+def _remove_services_from_initial_files(dir_path):
+    for installed_service in get_main_services_from_config():
+        _remove(os.path.join(dir_path, installed_service))
+        if dir_path == INITIAL_INSTALL_DIR:
+            update_yaml_file(INSTALLED_COMPONENTS, {installed_service: []})
+            update_yaml_file(INSTALLED_PACKAGES, {installed_service: []})
+    if _all_main_services_removed(dir_path):
+        _remove(dir_path)
+
+
+def _get_items_to_remove(items_file):
+    """
+    :param items_file: Can be either INSTALLED_COMPONENTS
+                       or INSTALLED_PACKAGES.
+    :return: A list of items (components or packages) that can be removed
+             without affecting the remaining services.
+
+    We use lists instead of sets to keep items' order.
+    """
+    items_dict = read_yaml_file(items_file)
+
+    def _get_items_list(lst, services_list):
+        for service_name in services_list:
+            for service_item in items_dict.get(service_name, []):
+                if service_item not in lst:
+                    lst.append(service_item)
+
+    potential_items_to_remove = []
+    remaining_items = []
+    removed_services = get_main_services_from_config()
```

Comment:
@YoniItzhak the `get_main_services_from_config` will always return `[DATABASE_SERVICE, QUEUE_SERVICE, MANAGER_SERVICE]` which means that `remaining_services` will always be empty list 
```
    remaining_services = ({DATABASE_SERVICE, QUEUE_SERVICE, MANAGER_SERVICE}
                          - set(removed_services))
```
`remaining_items` will be always empty in that case ?

Reply:
`get_services_from_config` returns the main services listed in the config. E.g. if the config.yaml file is of a postgresql instance, then `[DATABASE_SERVICE]` will return. In the case of an AIO, it is true that all services will return, and therefore `remoaining_item` will be empty. Which is OK, because after the removal there supposed to be no remaining items.